dask/dataframe/accessor.py

```diff
@@ -151,6 +155,47 @@ def getattr(obj, attr):
     def call(obj, attr, *args, **kwargs):
         return getattr(obj.cat, attr)(*args, **kwargs)
 
+    @property
+    def known(self):
+        """Whether the categories are fully known"""
+        return has_known_categories(self._series)
+
+    def as_known(self, **kwargs):
+        """Ensure the categories in this series are known.
+
+        If the categories are known, this is a no-op. If unknown, the
+        categories are computed, and a new series with known categories is
+        returned.
+
+        Parameters
+        ----------
+        kwargs
+            Keywords to pass on to the call to `compute`.
+        """
+        if self.known:
+            return self
+        categories = self._series.reduction(methods.cat_categories, M.unique)
+        categories = compute(categories, **kwargs)[0]
+        return self.set_categories(categories)
+
+    def as_unknown(self):
+        """Ensure the categories in this series are unknown"""
+        if not self.known:
+            return self._series
+        out = self._series.copy()
+        out._meta = clear_known_categories(out._meta)
+        return out
+
+    @property
+    def categories(self):
+        """The categories of this categorical.
+
+        If categories are unknown, an error is raised"""
+        if not has_known_categories(self._series._meta):
+            raise NotImplementedError("`df.cat.categories` with unknown "
```

Comment:
Could you point to `df.categorize` or `df.<column>.cat.set_categories` here. Like how you do in `get_dummies`?

It'd also be good to be consistent about naming here. If we use `df` as `dask.dataframe`, how about `df.column.cat.categories` instead of `df.cat.categories`.